art/attacks/evasion/graphite/utils.py

```diff
@@ -0,0 +1,627 @@
+        new_pt /= new_pt[2, 0]
+
+        if new_pt[0, 0] < min_x:
+            min_x = new_pt[0, 0]
+        if new_pt[0, 0] > max_x:
+            max_x = new_pt[0, 0]
+        if new_pt[1, 0] < min_y:
+            min_y = new_pt[1, 0]
+        if new_pt[1, 0] > max_y:
+            max_y = new_pt[1, 0]
+
+    if pts_flag:
+        if (max_x - min_x) / (max_y - min_y) < width / height:  # result is tall and narrow
+            diff_in_size = (max_y - min_y) / height * width - (max_x - min_x)
+            orig_size = max_y - min_y if width > height else (max_y - min_y) / height * width
+            crop_size = int(round(orig_size * (1.0 - crop_percent)))
+            y_off = -min_y - int(round(crop_percent / 2 * orig_size))
+            x_off = -min_x + int(round(diff_in_size / 2 - crop_percent / 2 * orig_size))
+
+        else:  # result is wide and short
+            diff_in_size = (max_x - min_x) / width * height - (max_y - min_y)
+            orig_size = max_x - min_x if height > width else (max_x - min_x) / width * height
+            crop_size = int(round(orig_size * (1.0 - crop_percent)))
+            x_off = -min_x - int(round(crop_percent / 2 * orig_size))
+            y_off = -min_y + int(round(diff_in_size / 2 - crop_percent / 2 * orig_size))
+
+        return x_off + crop_off_x * crop_size, y_off + crop_off_y * crop_size, crop_size
+
+    min_x -= (width * ratio - (max_x - min_x)) // 2
+    min_y -= (height * ratio - (max_y - min_y)) // 2
+
+    crop_size = int(round((1.0 - crop_percent) * min(width, height) * ratio))
+
+    return (
+        -min_x - int(round(crop_percent / 2 * width * ratio)),
+        -min_y - int(round(crop_percent / 2 * height * ratio)),
+        crop_size,
+    )
+
+
+def run_predictions(
+    estimator: "CLASSIFIER_TYPE", imgs: np.ndarray, target: int, batch_size: int, err_rate: bool = True
+) -> float:
+    """
+    Run model predictions over batch of input.
+    :param estimator: The model.
+    :param imgs: Batch of images with different transforms.
+    :param target: Target label.
+    :param batch_size: Size of batches to process.
+    :param err_rate: Whether to return a measure of error or a measure of transform-robutness.
+    :return: Either the error rate or transform-robustness depending on err_rate.
+    """
+    img_tensor = np.zeros((batch_size, imgs[0].shape[0], imgs[0].shape[1], imgs[0].shape[2]), dtype=np.float32)
+    num_successes = 0
+    tar_tensor = np.ones((batch_size)) * target
+
+    count = 0
+    for i, img in enumerate(imgs):
+        max_y = min(img.shape[0], img_tensor.shape[1])
+        max_x = min(img.shape[1], img_tensor.shape[2])
+        img_tensor[count, :max_y, :max_x, :] = img[:max_y, :max_x, :]
+        count += 1
+        if count == batch_size or i == len(imgs) - 1:
+            if count < batch_size:
+                img_tensor = img_tensor[:count, :, :, :]
+                tar_tensor = tar_tensor[:count]
+
+            if estimator.channels_first:
```

Comment:
I think you might want to use a different type or check to make sure that channels_first is an attribute